src/input/input_json.rs

```diff
@@ -144,7 +144,10 @@ impl<'a> Input<'a> for JsonInput {
             JsonInput::Uint(u) => Ok(EitherInt::U64(*u)),
             JsonInput::BigInt(b) => Ok(EitherInt::BigInt(b.clone())),
             JsonInput::Float(f) => float_as_int(self, *f),
-            JsonInput::String(str) => str_as_int(self, str),
+            JsonInput::String(str) => match str_as_int(self, str) {
+                Ok(i) => Ok(i),
+                Err(_) => str_as_int(self, &str.replace('_', "")),
```

Comment:
Some edge cases: trailing / leading underscores are not allowed in Python, nor is double-underscore.

```python
>>> int("5__0")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '5__0'
>>> int("5_0_")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '5_0_'
>>> int("_5_0")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '_5_0'
```

Reply:
Ugh annoying